app/lib/get-notification-data.js

```diff
@@ -0,0 +1,70 @@
+import apiClient from 'panoptes-client/lib/api-client';
+import talkClient from 'panoptes-client/lib/talk-client';
+
+function commentData(notification) {
+  return talkClient.type('comments').get(notification.source_id).then((comment) => {
+    return { notification, data: { comment } };
+  });
+}
+
+function requestData(notification) {
+  return apiClient.type('projects').get(notification.project_id).then((project) => {
+    return { notification, data: { projectName: project.display_name } };
+  });
+}
+
+function messageData(notification) {
+  return talkClient.type('messages').get(notification.source_id, { include: 'conversation' }).then((message) => {
+    return apiClient.type('users').get(message.user_id).then((messageUser) => {
+      return message.get('conversation').then((conversation) => {
+        return { notification, data: { message, conversation, messageUser } };
+      });
+    });
+  });
+}
+
+function moderationData(notification) {
+  return talkClient.type('moderations').get(notification.source_id).then((moderation) => {
+    const comment = moderation.target || moderation.destroyed_target;
+    const promises = [];
+    moderation.reports.map((report) => {
```

Comment:
`.map()` returns an array, so you can define `promises` as the result of that (just return each promise instead of using array.push)